library/src/blas3/geam_device.h

```diff
@@ -0,0 +1,124 @@
+/* ************************************************************************
+ * Copyright 2016 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+// general case for any alpha, beta, lda, ldb, ldc
+template<typename T>
+static __device__ void
+geam_device(
+    rocblas_operation transA, rocblas_operation transB,
+    rocblas_int m, rocblas_int n,
+    T alpha,
+    const T * __restrict__ A, rocblas_int lda,
+    const T * __restrict__ B, rocblas_int ldb,
+    T beta,
+          T *              C, rocblas_int ldc)
+{
+    rocblas_int tx = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    rocblas_int ty = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+
+    if (tx < m && ty < n)
+    {
+        if (transA == rocblas_operation_none && transB == rocblas_operation_none)
```

Comment:
I haven't checked if this is following the style elsewhere, but I would suggest only calculating the offsets under the conditions, and doing the actual work (loads, compute, store) outside of the conditions. Also, I'd consider using one fma and one multiply for potentially improved accuracy.

Reply:
The following commit contains improved code that addresses the comments 8a54224e8cf3034322b6c4b5ac6605f122078a21